### PR TITLE
tests: fix umount for snapd snap on fsck-on-boot test

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This error is faced running beta validation

+ [[ ubuntu-core-20-64 == ubuntu-core-20-arm-* ]]
+ umount '/run/mnt/ubuntu-seed/systems/*/snaps/snapd_*.snap'
umount: /run/mnt/ubuntu-seed/systems/*/snaps/snapd_*.snap: no mount
point specified.

Also, the test now is using the os-query to determine is the system is
running on arm

### CODE REVIEW

File: tests/core/fsck-on-boot/task.yaml [hunk 1]
```diff
@@ -47,22 +47,19 @@ execute: |
...
       umount /var/lib/snapd/seed
       umount /run/mnt/ubuntu-seed
-      if [[ "$SPREAD_SYSTEM" == ubuntu-core-20-arm-* ]]; then
-        umount /boot/uboot
+
+      # Refer to the core 20 gadgets for details:
+      # https://github.com/snapcore/pc-amd64-gadget/blob/20/gadget.yaml
+      # https://github.com/snapcore/pi-gadget/blob/20-arm64/gadget.yaml
+      if os.query is-arm; then
+          umount /boot/uboot
```
Comment (bboozzoo) [inline tests/core/fsck-on-boot/task.yaml:60]: The rest of the test looks to be using 2 spaces for indentation.

File: tests/core/fsck-on-boot/task.yaml [hunk 2]
```diff
@@ -47,22 +47,19 @@ execute: |
...
       # seeding anymore
 
-      # Refer to the core 20 gadgets for details:
-      # https://github.com/snapcore/pc-amd64-gadget/blob/20/gadget.yaml
-      # https://github.com/snapcore/pi-gadget/blob/20-arm64/gadget.yaml
       if mountpoint /run/mnt/snapd >/dev/null; then
-        if [[ "$SPREAD_SYSTEM" == ubuntu-core-20-arm-* ]]; then
           umount /run/mnt/ubuntu-seed/snaps/snapd_*.snap
-        else
-          umount /run/mnt/ubuntu-seed/systems/*/snaps/snapd_*.snap
```
Comment (bboozzoo) [inline tests/core/fsck-on-boot/task.yaml:57]: This path is removed for !arm.

File: tests/core/fsck-on-boot/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/core/fsck-on-boot/task.yaml]: updated

File: tests/core/fsck-on-boot/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/core/fsck-on-boot/task.yaml:57]: yes, that's why I updated the test.

**Review by bboozzoo — APPROVED** [review]

**Review by stolowski — APPROVED** [review]
LGTM

**Pull request merged**